pkg/plugins/policies/matchers/match.go

```diff
@@ -13,44 +13,102 @@ import (
 	xds_context "github.com/kumahq/kuma/pkg/xds/context"
 )
 
-type ResourceSpecWithTargetRef interface {
-	core_model.ResourceSpec
-	GetTargetRef() *common_proto.TargetRef
-}
-
 // MatchedPolicies match policies using the standard matchers using targetRef (madr-005)
 func MatchedPolicies(rType core_model.ResourceType, dpp *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error) {
+	result := core_xds.TypedMatchingPolicies{
+		Type: rType,
+	}
+
 	policies := resources.ListOrEmpty(rType)
 
-	matchedPolicies := []core_model.Resource{}
+	matchedPoliciesByInbound := map[mesh_proto.InboundInterface][]core_model.Resource{}
```

Comment:
Btw I've noticed that `mesh_proto.InboundInterface` is not the nicest to use as a map key because it has a bunch of fields that has nothing to do with matching. Should we have a simple type with just address/port?